packages/store-sync/src/createStoreSync.ts

```diff
@@ -344,23 +346,49 @@ export async function createStoreSync({
       // instead of sending the next request only when the previous one completed.
       mergeMap(async (blocks) => {
         for (const block of blocks) {
-          const txs = block.logs.map((op) => op.transactionHash);
-          // If the transaction caused a log, it must have succeeded
-          if (txs.includes(tx)) {
-            return { blockNumber: block.blockNumber, status: "success" as const, transactionHash: tx };
+          // If storage adapter block had Store event logs associated with tx, it must have succeeded.
+          if (block.logs.some((log) => log.transactionHash === tx)) {
+            return { blockNumber: block.blockNumber, status: "success", transactionHash: tx } as const;
           }
         }
 
         try {
           const lastBlock = blocks[0];
           debug("fetching tx receipt for block", lastBlock.blockNumber);
-          const { status, blockNumber, transactionHash } = await getAction(
-            publicClient,
-            getTransactionReceipt,
-            "getTransactionReceipt",
-          )({ hash: tx });
-          if (lastBlock.blockNumber >= blockNumber) {
-            return { status, blockNumber, transactionHash };
+          const receipt = await getAction(publicClient, getTransactionReceipt, "getTransactionReceipt")({ hash: tx });
+
+          if (lastBlock.blockNumber >= receipt.blockNumber) {
```

Comment:
why do we need this check? i see we also had it before, before we were not returning anything if the check didn't pass, now it seems like we're still returning the receipt in that case (but are not performing the user op checks)?